helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -781,8 +845,15 @@ void reset(boolean isShutdown) {
           if (isShutdown) {
             _batchCallbackProcessor.shutdown();
             _batchCallbackProcessor = null;
+            if (_periodicRefreshExecutor != null) {
+              _periodicRefreshExecutor.shutdownNow();
+            }
           } else {
             _batchCallbackProcessor.resetEventQueue();
+            if (_scheduledRefreshFuture != null) {
+              _periodicRefreshExecutor.shutdownNow();
+              initRefreshTask();
```

Comment:
The next event would be FINALIZE, so even the periodic task is triggered, it won't really invoke the handling logic successfully. I think we shall not to init here.

Shall we do it in the init() call?